### ISSUE TITLE
Using torch.log when apply KLDivLoss.

### ISSUE THREAD

**Issue by miangangzhen** [michaal94/torch_DCEC#1]
Wondering why using torch.log when apply KLDivLoss.

Your code:
```python
loss_clust = gamma *criteria[1](torch.log(clusters), tar_dist) / batch
```
My code:
```python
loss = criterion(y_hat_batch, torch.from_numpy(y_batch))
```

My code doesn't work, but yours did. I can't figure out what's torch.log role?

**Comment by michaal94** [comment]
It is because PyTorch expects logprob as the input for KL loss.
Check: [link](https://pytorch.org/docs/stable/nn.html?highlight=loss#torch.nn.KLDivLoss)
